test/integration/raft_test.lua

```diff
@@ -0,0 +1,474 @@
+end
+
+g.test_disable_raft_failover = function()
+    local assertions = g.cluster:server('storage-1'):exec(function()
+        return {
+            #box.ctl.on_election(),
+            box.cfg.election_mode,
+            box.info.election.state,
+        }
+    end)
+    t.assert_equals(assertions, {
+        1,
+        'candidate',
+        'leader',
+    })
+
+    h.retrying({}, function ()
+        set_failover_params{mode = 'disabled'}
+    end)
+
+    local assertions = g.cluster:server('router-1'):exec(function()
+        return {
+            #box.ctl.on_election(),
+            box.cfg.election_mode,
+            box.info.election.state,
+        }
+    end)
+    t.assert_equals(assertions, {
+        0,
+        'off',
+        'follower',
+    })
+end
+
+g.after_test('test_disable_raft_failover', function()
+    h.retrying({}, function ()
+        set_failover_params{mode = 'raft'}
+    end)
+end)
+
+
+g.test_graphql_errors = function()
+    t.assert_error_msg_contains(
+        'failover.election_timeout must be non-negative',
+        set_failover_params,
+        {
+            election_timeout = -1,
+        }
+    )
+    t.assert_error_msg_contains(
+        'failover.synchro_timeout must be non-negative',
+        set_failover_params,
+        {
+            synchro_timeout = -1,
+        }
+    )
+    t.assert_error_msg_contains(
+        'failover.replication_timeout must be non-negative',
+        set_failover_params,
+        {
+            replication_timeout = -1,
+        }
+    )
+end
+
+g.before_test('test_bucket_ref_on_replica_prevent_bucket_move', function()
+    g.cluster.main_server:exec(function()
+        local vshard_router = require('vshard.router')
+        local key = 'key'
+        local bucket_id = vshard_router:bucket_id_strcrc32(key)
+        vshard_router.callrw(bucket_id, 'box.space.test:insert',
+                {{bucket_id, key, {}}})
+    end)
+end)
+
+-- see https://github.com/tarantool/vshard/issues/173 for details
+g.test_bucket_ref_on_replica_prevent_bucket_move = function()
+    t.xfail('Test fails until tarantool/vshard#173 will be fixed')
+    -- ref bucket on replica
+    local some_bucket_id = g.cluster:server('storage-2'):exec(function()
+        assert(box.info.ro)
+        local some_bucket_id = box.space.test:pairs():nth(1).bucket_id
+        local vshard_storage = require('vshard.storage')
+        vshard_storage.bucket_ref(some_bucket_id, 'read')
+        return some_bucket_id
+    end)
+
+    g.cluster.main_server:setup_replicaset({
+        weight = 1,
+        uuid = single_replicaset_uuid,
+    })
+
+    -- send bucket to another storage
+    local bucket_counts = g.cluster:server('storage-1'):exec(function(bucket_id, replicaset_uuid)
+        assert(box.info.ro ~= true)
+        local vshard_storage = require('vshard.storage')
+        vshard_storage.bucket_send(bucket_id, replicaset_uuid)
+        -- wait until sending
+        require('fiber').sleep(0.5)
```

Comment:
Why not "wait_for"? Sleeps in code are quite unreliable. Especially on CI where a lot of tasks/containers are run simultaneously.